Source/WebCore/workers/service/context/ServiceWorkerThread.cpp

```diff
@@ -246,6 +246,9 @@ void ServiceWorkerThread::queueTaskToFirePushEvent(std::optional<Vector<uint8_t>
             }
 
             bool showedNotification = !serviceWorkerGlobalScope->hasPendingSilentPushEvent();
+            serviceWorkerGlobalScope->setHasPendingSilentPushEvent(false);
+            if (!showedNotification)
+                serviceWorkerGlobalScope->addConsoleMessage(MessageSource::Storage, MessageLevel::Error, "Push event ended without showing any notification"_s, 0);
```

Comment:
I wonder if we should be explicit about the consequences here. Like "Push event ended without showing any notification. After three such silent push events, the browser will remove your push subscription."

Reply:
I'll mention the risk of the push subscription removal, but not too precise (I do not want to change the message if we change the rule).